spec/mongo/collection/view_spec.rb

```diff
@@ -18,6 +18,149 @@
+      end
+
+      it 'sets the modifiers' do
+        expect(view.instance_variable_get(:@modifiers)).to eq(modifiers)
+      end
+
+      it 'removes the modifiers from the selector' do
+        expect(view.selector).to eq(parsed_selector)
+      end
+
+      it 'creates the correct query selector' do
+        expect(view.send(:query_spec)[:selector]).to eq(query_selector)
+      end
+    end
+
+    context 'when a modifiers document is provided in the options' do
+
+      let(:selector) do
+        { a: 1 }
+      end
+
+      let(:options) do
+        { :modifiers => { :$someMod => 100 } }
+      end
+
+      let(:modifiers) do
+        { :$someMod => 100 }
+      end
+
+      let(:parsed_selector) do
+        { a: 1 }
+      end
+
+      let(:query_selector) do
+        { :$query => { a: 1 }, :$someMod => 100 }
+      end
+
+      it 'sets the modifiers' do
+        expect(view.instance_variable_get(:@modifiers)).to eq(modifiers)
+      end
+
+      it 'removes the modifiers from the selector' do
+        expect(view.selector).to eq(parsed_selector)
+      end
+
+      it 'creates the correct query selector' do
+        expect(view.send(:query_spec)[:selector]).to eq(query_selector)
+      end
+
+      context 'when modifiers and options are both provided' do
+
+        let(:selector) do
+          { a: 1 }
+        end
+
+        let(:options) do
+          { :sort =>  { a: Mongo::Index::ASCENDING }, :modifiers => { :$orderby => { a: Mongo::Index::DESCENDING } } }
+        end
+
+        let(:modifiers) do
+          { :$orderby => { a: Mongo::Index::ASCENDING } }
+        end
+
+        let(:parsed_selector) do
+          { a: 1 }
+        end
+
+        let(:query_selector) do
+          { :$query => { a: 1 }, :$orderby => { a: Mongo::Index::ASCENDING } }
+        end
+
+        it 'sets the modifiers' do
+          expect(view.instance_variable_get(:@modifiers)).to eq(modifiers)
+        end
+
+        it 'removes the modifiers from the selector' do
+          expect(view.selector).to eq(parsed_selector)
+        end
+
+        it 'creates the correct query selector' do
+          expect(view.send(:query_spec)[:selector]).to eq(query_selector)
+        end
+
+        context 'when modifiers, options and a query modifier are provided' do
+
+          let(:selector) do
+            { b: 2, :$query => { a: 1 }, :$someMod => 100 }
+          end
+
+          let(:options) do
+            { :sort =>  { a: Mongo::Index::ASCENDING }, :modifiers => { :$someMod => true, :$orderby => { a: Mongo::Index::DESCENDING } } }
+          end
+
+          let(:modifiers) do
+            { :$query => { a: 1 }, :$someMod => 100, :$orderby => { a: Mongo::Index::ASCENDING } }
+          end
+
+          let(:parsed_selector) do
+            { b: 2 }
+          end
+
+          let(:query_selector) do
+            { :$query => { a: 1 }, :$someMod => 100, :$orderby => { a: Mongo::Index::ASCENDING } }
+          end
```

Comment:
@behackett I'm assuming that if you provide a $query doc in the selector, it'll use that as the selector, not whatever else is in there. I.e., in this case, { a: 1 } is the selector, not { b: 2 }